cdap-app-fabric/src/main/java/co/cask/cdap/internal/bootstrap/BootstrapStep.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package co.cask.cdap.internal.bootstrap;
+
+import com.google.gson.JsonObject;
+
+/**
+ * Definition for a single step in a bootstrap process. Defines what operation to perform, whether it should be executed
+ * every time CDAP starts up or just once, and any arguments required to perform the operation.
+ */
+public class BootstrapStep {
+  private final String label;
```

Comment:
there is no requirement for label to be unique right ? just that the user can get unambiguous response if they follow unique labeling.

Reply:
that's correct